editor/plugins/path_3d_editor_plugin.cpp

```diff
@@ -288,70 +288,74 @@ void Path3DGizmo::redraw() {
 		return;
 	}
 
-	real_t interval = 0.1;
 	const real_t length = c->get_baked_length();
-
 	// 1. Draw curve and bones.
 	if (length > CMP_EPSILON) {
-		const int sample_count = int(length / interval) + 2;
-		interval = length / (sample_count - 1); // Recalculate real interval length.
-
-		Vector<Transform3D> frames;
-		frames.resize(sample_count);
-
-		{
-			Transform3D *w = frames.ptrw();
-
-			for (int i = 0; i < sample_count; i++) {
-				w[i] = c->sample_baked_with_rotation(i * interval, true, true);
+		const int point_count = c->get_point_count();
+		const int sample_count = c->get_debug_preview_resolution() + 2;
+		for (int point = 0; point < point_count - 1; point++) {
+			const real_t section_length = c->get_baked_distance_at_point(point + 1) - c->get_baked_distance_at_point(point);
+			const real_t interval = section_length / (sample_count - 1); // Recalculate real interval length of section.
+			Vector<Transform3D> frames;
+			frames.resize(sample_count);
+
+			real_t offset;
+			point == 0 ? offset = 0 : offset = c->get_baked_distance_at_point(point);
```

Comment:
```suggestion
			real_t offset = point == 0 ? 0 : c->get_baked_distance_at_point(point);
```
Don't use assignment inside a ternary